[zh] docs/setup/production-environment/container-runtimes.md (#36045)
kubernetes/website: content/zh-cn/docs/setup/production-environment/container-runtimes.md

[announced](/blog/2020/12/08/kubernetes-1-20-release-announcement/#dockershim-deprecation)
as part of the v1.20 release).
-->

> kubernetes v1.24 之前的版本，使用 **dockershim** 组件直接整合了 Docker Engine 。

Reviewer: ```suggestion
Kubernetes v1.24 之前的版本，使用 **dockershim** 组件直接整合了 Docker Engine 。
```

affect you. To learn about migrating from using dockershim, see
[Migrating from dockershim](/docs/tasks/administer-cluster/migrating-from-dockershim/).
-->

> 你可以阅读[检查弃用 Dockershim 是否会影响你](/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/check-if-dockershim-deprecation-affects-you/)

Reviewer: ```suggestion
你可以阅读[检查移除 Dockershim 是否对你有影响](/zh-cn/docs/tasks/administer-cluster/migrating-from-dockershim/check-if-dockershim-removal-affects-you/)
```

单个 cgroup 管理器将简化分配资源的视图，并且默认情况下将对可用资源和使用
中的资源具有更一致的视图。
当有两个管理器共存于一个系统中时，最终将对这些资源产生两种视图。
在这方面，人们已经报告过一些示例，某些节点的 kubelet 和 docker 使用

> `cgroupfs`，而节点上运行的其余进程则使用 `systemd`，这类节点在资源压力下

Reviewer: ```suggestion
`cgroupfs`，而节点上运行的其余进程则使用 `systemd`，
```

> 会变得不稳定。

Reviewer: ```suggestion
这类节点在资源压力下会变得不稳定。
```

Author: Thanks for your suggestions.

别的 cgroup 驱动，当为现有 Pods 重新创建 PodSandbox 时会产生错误。
重启 kubelet 也可能无法解决此类问题。

> 除非你有切实可行的自动化方案，使用其他已更新配置的节点来替换该节点，

Reviewer 2: 改错了

Author: Has been modified

-->
## CRI 版本支持 {#cri-versions}

> 你的容器运行时必须至少支持v1alpha2 版本的容器运行时接口。

Reviewer: ```suggestion
你的容器运行时必须至少支持 v1alpha2 版本的容器运行时接口。
```

> Kubernetes {{< skew currentVersion >}} 默认使用 v1 版本的 CRI API。如果容器运行时不支持 v1 版本的API，

Reviewer: ```suggestion
Kubernetes {{< skew currentVersion >}} 默认使用 v1 版本的 CRI API。如果容器运行时不支持 v1 版本的 API，
```

> 则 kubelet 会回退到使用（已弃用的）v1alpha2 版本的API。

Reviewer: ```suggestion
则 kubelet 会回退到使用（已弃用的）v1alpha2 版本的 API。
```

-->
本节概述了使用 containerd 作为 CRI 运行时的必要步骤。

> 使用以下命令在系统上安装 Containerd ：

Reviewer: ```suggestion
使用以下命令在系统上安装 Containerd：
```
This space is not necessary. Same below.

如果你从软件包（例如，RPM 或者 `.deb`）中安装 containerd，你可能会发现其中默认禁止了
CRI 集成插件。

> 你需要启用 CRI 支持才能在 Kubernetes 集群中使用 containerd 。

Reviewer: The space is not necessary.

Reviewer: Not modified but click Resolved...

要确保 `cri` 没有出现在 `/etc/containerd/config.toml` 文件中 `disabled_plugins`
列表内。如果你更改了这个文件，也请记得要重启 `containerd`。
{{< /note >}}

<!--
If you apply this change, make sure to restart containerd:
-->

> 如果你应用此更改，请确保重新启动 containerd ：

Reviewer: The space is not necessary.

Reviewer: Not modified but click Resolved...

cgroup driver configuration of the kubelet (usually done via kubeadm) and CRI-O
in sync.
-->

> 你还应该注意当使用CRI-O时，并且CRI-O 的 cgroup 设置为 `cgroupfs` 时，必须将`conmon_cgroup` 设置为值 `pod`。

Reviewer: ```suggestion
你还应该注意当使用 CRI-O 时，并且 CRI-O 的 cgroup 设置为 `cgroupfs` 时，必须将 `conmon_cgroup` 设置为值 `pod`。
```
